### PR TITLE
add stream logger

### PR DESCRIPTION
**Pull request by m-pilia**
Required for https://github.com/simeks/deform/pull/37

### CODE REVIEW

**Comment by m-pilia** [comment]
I am not sure how this change is creating a segfault in `test_io.cpp`, that technically should have nothing to do with these edited files at all... I'm kinda tired so maybe I am missing something dumb, but I am having a look at this asap.

File: src/stk/common/log.cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ namespace
 
     void log_write(stk::LogLevel level, const char* msg)
     {
-        if (!_logger_data->silent) {
+        if (nullptr == _logger_data || !_logger_data->silent) {
```
Comment (m-pilia) [inline src/stk/common/log.cpp]: Forget it, it's just me dumber than usual today...

**Comment by simeks** [comment]
Great! I knew the logging system would come in handy one day 😃

**Comment by m-pilia** [comment]
Yeah, the logging system is well designed, so it made the addition very easy.

**Pull request merged**